### ISSUE TITLE
Internal server errors when enabling Prometheus metrics on v1.5.0

### ISSUE THREAD

**Issue by sebastien-prudhomme** [containous/traefik#2760]
Migrating from v1.4.6 (with no Prometheus metrics).
Running on official Docker image.

Previous configuration was:

```toml
checkNewVersion = false
defaultEntryPoints = ["http", "https"]
[accessLog]
[consulCatalog]
  constraints = ["tag==http"]
  domain = "example.com"
[entryPoints]
  [entryPoints.http]
    address = ":80"
    compress = true
  [entryPoints.https]
    address = ":443"
    compress = true
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = "/etc/traefik/traefik.crt"
      keyFile = "/etc/traefik/traefik.key"
[web]
```

New configuration on v1.5.0 with Prometheus metrics:

```toml
checkNewVersion = false
defaultEntryPoints = ["http", "https"]
[accessLog]
[api]
[consulCatalog]
  constraints = ["tag==http"]
  domain = "example.com"
[entryPoints]
  [entryPoints.http]
    address = ":80"
    compress = true
  [entryPoints.https]
    address = ":443"
    compress = true
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = "/etc/traefik/traefik.crt"
      keyFile = "/etc/traefik/traefik.key"
[metrics]
  [metrics.prometheus]
[ping]
```

Access to http://localhost:8080/api/ works fine.
Access to http://localhost:8080/dashboard/ works fine.
Access to http://localhost:8080/metrics/ works fine.
Access to http://localhost:8080/ping/ works fine.

Access to applications behind Traefik no more work (500 Internal Server Error). A lot of these lines in the logs:

```
time="2018-01-24T16:36:45Z" level=error msg="Recovered from panic in http handler: interface conversion: *negroni.responseWriter is not http.CloseNotifier: missing method CloseNotify" 
```

If i remove the two "metrics" lines in configuration, all works fine.

**Comment by sebastien-prudhomme** [comment]
If i disable compression like in #2628, all works fine

**Comment by traefiker** [comment]
Closed by #2766.
